### ISSUE TITLE
Incorrect disk size and usage detection blocks e2e from starting with Kops

### ISSUE THREAD

**Issue by hakman** [kubernetes/kubernetes#91982]
**Which jobs are failing**:
e2e-kops-aws-k8s-latest

**Which test(s) are failing**:
Tests cannot start at all because critical pods cannot be scheuled

**Since when has it been failing**:
We noticed this was happening last week and investigations pointed out eviction manager as the root cause. We disabled out of resource handling for imagefs and things are working again.

We didn't notice this sooner because kubelet kept being restarted, which hid the issue. This was fixed by https://github.com/kubernetes/kubernetes/pull/89589.

**Testgrid link**:
https://gcsweb.k8s.io/gcs/kubernetes-jenkins/logs/e2e-kops-aws-k8s-latest/1269848278476787712/

**Reason for failure**:
Relevant log from the Kubelet:
```
W0608 04:40:31.222453    4996 eviction_manager.go:339] eviction manager: attempting to reclaim ephemeral-storage
I0608 04:40:31.222488    4996 container_gc.go:85] attempting to delete unused containers
I0608 04:40:31.225274    4996 image_gc_manager.go:322] attempting to delete unused images
I0608 04:40:31.231317    4996 eviction_manager.go:350] eviction manager: must evict pod(s) to reclaim ephemeral-storage
I0608 04:40:31.231341    4996 eviction_manager.go:368] eviction manager: pods ranked for eviction: kube-proxy-ip-172-20-42-239.ap-northeast-2.compute.internal_kube-system(98b80b03aaa21bd38ed134d9616a3730)
E0608 04:40:31.231362    4996 eviction_manager.go:569] eviction manager: cannot evict a critical pod kube-proxy-ip-172-20-42-239.ap-northeast-2.compute.internal_kube-system(98b80b03aaa21bd38ed134d9616a3730)
I0608 04:40:31.231370    4996 eviction_manager.go:391] eviction manager: unable to evict any pods from the node
I0608 04:40:40.627941    4996 image_gc_manager.go:305] [imageGCManager]: Disk usage on image filesystem is at 100% which is over the high threshold (85%). Trying to free 3774873 bytes down to the low threshold (80%).
E0608 04:40:40.631282    4996 kubelet.go:1231] Image garbage collection failed multiple times in a row: failed to garbage collect required amount of images. Wanted to free 3774873 bytes, but freed 0 bytes
I0608 04:40:40.631730    4996 kubelet_getters.go:173] "Pod status updated" pod="kube-system/kube-proxy-ip-172-20-42-239.ap-northeast-2.compute.internal" status="Running"
```

**Anything else we need to know**:
Corresponding issue in Kops: https://github.com/kubernetes/kops/issues/9301

Observations:
* Kubernetes 1.18 works OK
* using containerd 1.3.4 as container runtime works OK
* restarting kubelet.service fixes the issue
* using Docker 19.03.x + Kubernetes 1.19 doesn't work

**Comment by hakman** [comment]
@dashpole I think I tracked the issue to a bug in cAdvisor. I guess we will continue the discussion there.
https://github.com/google/cadvisor/pull/2586

**Comment by hakman** [comment]
Fixed by #91366.
/close
